app/models/ncNews.models.js

```diff
@@ -34,20 +34,26 @@ exports.fetchArticleById = (article_id) => {
     });
 }
 
-exports.fetchArticles = () => {
-  return db
-    .query(
-      `
-      SELECT articles.author, articles.title, articles.article_id, articles.topic, articles.created_at, articles.votes, articles.article_img_url, COUNT(comments.article_id) AS comment_count FROM articles
-      LEFT JOIN comments ON comments.article_id = articles.article_id GROUP BY articles.article_id
-      ORDER BY created_at DESC;
-      `)
-    .then(({ rows }) => {
-      if (rows.length === 0) {
-        return Promise.reject({status: 404, msg: "Not Found"})
-      }
+
+exports.fetchArticles = (topic) => {
+
+  let queryStr  =  `
+  SELECT articles.author, articles.title, articles.article_id, articles.topic, articles.created_at, articles.votes, articles.article_img_url, COUNT(comments.article_id) AS comment_count FROM articles
+  LEFT JOIN comments ON comments.article_id = articles.article_id
+  `;
+
+  const queryParameters = [];
+
+  if (topic) {
+    queryStr += " WHERE topic = $1 GROUP BY articles.article_id ORDER BY created_at DESC"
+    queryParameters.push(topic)
+  } else {
+    queryStr += " GROUP BY articles.article_id ORDER BY created_at DESC";
+  }
+
```

Comment:
You can get rid of the else here and just always add the text following GROUP BY instead of repeating it. 

If there is a topic, just add the WHERE clause 

always add GROUP BY etc.

Reply:
Hi Christian,
Thanks for your feedback on task 11.
Just I am not sure how to do this! I though WHERE always have to come before GROUP BY.
Would be great if you get a chance to clarify this :)

Reply:
Thanks for clarification, done :)